commands/commands.go

```diff
@@ -644,3 +653,19 @@ func getCertPathInfo(c *cli.Context) libmachine.CertPathInfo {
 		ClientKeyPath:  clientKeyPath,
 	}
 }
+
+func detectShell() (string, error) {
+	// attempt to get the SHELL env var
+	shell := filepath.Base(os.Getenv("SHELL"))
+	// none detected; check for windows env
+	if shell == "." && os.Getenv("windir") != "" {
```

Comment:
I think this check should be:

    if runtime.GOOS == "windows"

although  `shell == "." && os.Getenv("windir") != ""` implies the same thing, you should be calling filepath.Base after `if windows` b/c it's not used in windows case at all.

also don't rely on `windir` :) People can run docker.exe with unsetting the env. GOOS is safe to use.